File: sum_strings_as_numbers.py
from itertools import zip_longest

def sum_strings(x, y):
    x, y = x.lstrip("0"), y.lstrip("0")
    carryover = 0
    if len(x) < len(y):
        x = x.zfill(len(y))
    elif len(x) > len(y):
        y = y.zfill(len(x))
    nums = list(reversed(list(zip_longest(x, y, fillvalue=0))))
    sol = ""

    for index, (val1, val2) in enumerate(nums):
        if index == (len(nums) - 1):
            sol += "".join(reversed(str(int(val1) + int(val2) + carryover)))
        else:
            sol += str(int(val1) + int(val2) + carryover)[-1]
            carryover = (int(val1) + int(val2) + carryover) // 10

    return "".join(reversed(list(sol))) if not "".join(reversed(list(sol))) == "" else "0"

print(sum_strings("00103", "08567"))

File: sum_strings_as_numbers.js
function sumStrings(a, b) {
  if (a === "") {
    a = BigInt(0);
  } else {
    a = BigInt(a);
  }
  if (b === "") {
    b = BigInt(0);
  } else {
    b = BigInt(b);
  }

  return (a + b).toString();
}

console.log(
  sumStrings("712569312664357328695151392", "8100824045303269669937")
);
